kernel/src/syscall/mount.rs

```diff
@@ -233,6 +233,13 @@ fn create_overlayfs(data: &str, ctx: &Context) -> Result<Arc<OverlayFS>> {
         .collect();
     let work = fs.lookup(&FsPath::new(AT_FDCWD, work)?)?;
 
+    if !Arc::ptr_eq(upper.mount_node(), work.mount_node()) {
```

Comment:
This check should be added to `OverlayFS::new` so that no users of `OverlayFS` may create invalid instances.

I am pretty sure that there should be more sanity checks added to the constructor method. Just to name a few:
* `upper`, `lower`, and `work` are  directories;
* `workdir` is empty.

Reply:
Makes sense, I'll move the checks to the construction method.